src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -711,34 +711,42 @@ def import_aaz_by_profile(profile, module_name):
 
 
 def generate_ssh_keys_ed25519(private_key_filepath, public_key_filepath):
+    def _open(filename, mode):
+        return os.open(filename, flags=os.O_WRONLY | os.O_TRUNC | os.O_CREAT, mode=mode)
+
     from cryptography.hazmat.primitives import serialization
     from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
 
     ssh_dir = os.path.dirname(private_key_filepath)
     if not os.path.exists(ssh_dir):
-        os.makedirs(ssh_dir)
-        os.chmod(ssh_dir, 0o700)
+        os.makedirs(ssh_dir, mode=0o700)
 
-    private_key = Ed25519PrivateKey.generate()
-    public_key = private_key.public_key()
-    private_bytes = private_key.private_bytes(
-        encoding=serialization.Encoding.PEM,
-        format=serialization.PrivateFormat.OpenSSH,
-        encryption_algorithm=serialization.NoEncryption()
-    )
-
-    with os.fdopen(os.open(private_key_filepath, flags=os.O_WRONLY | os.O_TRUNC | os.O_CREAT, mode=384, ), "w", ) as f:
-        f.write(
-            private_bytes.decode()
+    if os.path.isfile(private_key_filepath):
+        # Try to use existing private key if it exists.
+        with open(private_key_filepath, "rb") as f:
+            private_bytes = f.read()
+        private_key = serialization.load_pem_private_key(private_bytes, password=None)
+        logger.warning("Private SSH key file '%s' was found in the directory: '%s'. "
+                       "A paired public key file '%s' will be generated.",
+                       private_key_filepath, ssh_dir, public_key_filepath)
+
+    else:
+        private_key = Ed25519PrivateKey.generate()
+        private_bytes = private_key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.PKCS8,
```

Comment:
May I ask why is `PrivateFormat.PKCS8` used here instead of `PrivateFormat.OpenSSH`?

Reply:
because it will throw an exception when loading the private key with format PrivateFormat.OpenSSH

Reply:
It will throw an exception when loading a private key with format OpenSSH by calling serialization.load_pem_private_key() function.
Alternatively, use serialization.load_ssh_private_key() function to load OpenSSH private key.